http4k-format/moshi/src/main/kotlin/org/http4k/format/MoshiNode.kt

```diff
@@ -0,0 +1,50 @@
+package org.http4k.format
+
+import java.math.BigDecimal
+import java.math.BigInteger
+
+sealed interface MoshiNode {
+    companion object
+}
+
+data class MoshiArray(val elements: List<MoshiNode>): MoshiNode
+data class MoshiObject(val attributes: Map<String, MoshiNode>): MoshiNode
+sealed interface MoshiPrimitive: MoshiNode
+data class MoshiString(val value: String): MoshiPrimitive
+data class MoshiInteger(val value: BigInteger): MoshiPrimitive
+data class MoshiDecimal(val value: BigDecimal): MoshiPrimitive
+data class MoshiBoolean(val value: Boolean): MoshiPrimitive
+object MoshiNull: MoshiPrimitive
+
+fun MoshiNode.unwrap(): Any? = when(this) {
+    is MoshiArray -> elements.map { it.unwrap() }
+    is MoshiObject -> attributes.mapValues { (_, value) -> value.unwrap() }
+    is MoshiString -> value
+    is MoshiInteger -> value
+    is MoshiDecimal -> value
+    is MoshiNull -> null
+    is MoshiBoolean -> value
+}
+
+fun MoshiNode.Companion.wrap(obj: Any?): MoshiNode = when(obj) {
+    null -> MoshiNull
+    is Iterable<*> -> obj
+        .map { wrap(it) }
+        .toList()
+        .let { MoshiArray(it) }
+    is Map<*, *> -> obj
+        .mapKeys { (key, _) -> key.toString() }
+        .mapValues { (_, value) -> wrap(value) }
+        .let { MoshiObject(it) }
+    is Number -> {
```

Comment:
Not sure sure about this, because it would convert an intentional `123.0` to an integer.  But with the way Moshi works, it's either this, or convert all numbers to decimal.